fair_mango/metrics/base.py

```diff
@@ -212,7 +213,9 @@ def __init__(
     def __call__(self): ...
 
 
-def difference(result_per_groups: np.ndarray) -> dict[tuple, np.ndarray[float]]:
+def calculate_disparity(
+    result_per_groups: np.ndarray, method: Literal["difference", "ratio"]
+) -> dict[tuple, np.ndarray[float]]:
```

Comment:
I used generator to get the disparity values. However, I don't see the benefit since the operation is done once on a numpy array (which directly does it element by element). 

Also, we are dealing with one value or two for the most part so the memory usage difference will be negligible too.